RXPatchLib/WebPatchSource.cs

```diff
@@ -100,6 +100,10 @@ private void UnlockDownload()
         {
             string filePath = GetSystemPath(subPath);
             var guid = Guid.NewGuid();
+            var thisPatchServer = _patcher.UpdateServerSelector.GetNextAvailableServerEntry();

```

Comment:
We shouldn't move this up here, because if we do then failover will no longer function because a new mirror will never be selected.

Reply:
Doesn't really matter that it's up there, can be fixed by asking for a new host during the catch.